src/kernel_abi.h

```diff
@@ -592,6 +593,20 @@ struct BaseArch : public wordsize, public FcntlConstants {
   };
   RR_VERIFY_TYPE(user_desc);
 
+  // TODO(dannas): When to we need typedefs for types?
+  // sizeof(int) will always be the x86_64 size. Guess I should use signed_int
+  // for pid? But __u32 should be the same on 32bit and 64 bit platforms, right?
+  struct cap_user_header_t {
+    __u32 version;
+    int pid;
+  };
+
+  struct cap_user_data_t {
+    __u32 effective;
+    __u32 permitted;
+    __u32 inheritable;
+  };
```

Comment:
I like what you've written here. I like to have the kernel_abi definitions match the actual header contents as closely as possible. So I guess you just need to add a typedef for `__u32` to `BaseArch`.